docs/source/cc_service.md

```diff
@@ -0,0 +1,297 @@
+* **cert_path**  - path to the certificate file. This path is relative to the “release directory” (see “release” below). Must be specified if client_auth_required is true. Ignored if tls_required is false.
+* **root_cert_path**  - path to the root cert for authenticating the server. Required when tls_required is set to true.
+
+For example:
+```json
+{
+  "address": "your.chaincode.host.com:9999",
+  "dial_timeout": "10s",
+  "tls_required": true,
+  "client_auth_required": "true",
+  "key_path": "path/rooted/in/release/directory/key.pem",
+  "cert_path": "path/rooted/in/release/directory/key.pem",
+  "root_cert_path": "path/rooted/in/release/directory/key.pem"
+}
+```
+
+**Note:** The TLS files can be placed anywhere in the `code.tar.gz` archive because the directory of the contents of the `.targz` will be provided to the external chaincode builder scripts. The `/bin/release` script, described later in this document, should move the files to the appropriate paths relative to the release directory.
+
+### metadata.json requirements
+
+The metadata.json file contains three keys:
+
+* **path** - not used by the chaincode as an external service.
+* **type** - set to `external` to specify this is an external chaincode.
+* **label** - name of the chaincode as used in the chaincode lifecycle model.
+
+For example:
+
+```json
+{"path":"","type":"external","label":"mycc"}
+```
+
+## Configuring a peer to process external chaincode
+
+If you have reviewed the external build and chaincode launcher documentation then these steps will be familiar to you. We leverage those scripts to define the external chaincode service information. These scripts are located on the peer file system that is accessible to the peer process from the `core.yaml` file in the `externalBuilders` element under the `chaincode` stanza. An example is included in the steps which follow.
+
+### Create the set of external builder and launcher scripts on the peer
+
+In order to configure chaincode as an external service, you must use the scripts as follows:
+* **detect** - examines the chaincode package and accept if metadata.json `type` is set to `external`.
+* **build** - builds the chaincode and places the build artifacts in the `BUILD_OUTPUT_DIR`. For chaincode as an external service, the peer is not does not build the chaincode. Instead, the script extracts the chaincode endpoint information in the `connection.json` file and places it in the specified location.
+* **release** - copies the built artifacts (in our case the `connection.json` file) to a specified release location.
+* **run** - Not required when using chaincode as an external service if the runtime of chaincode running on the external server is manged externally and the chaincode need not be started from this script.
+
+The scripts are required to exist in the peer `/bin` directory:
+```
+    <fully qualified path on the peer's env>
+    └── bin
+        ├── build
+        ├── detect
+        ├── release
+        └── run
```

Comment:
We don't need a run binary, should be fine to omit it.

Reply:
done